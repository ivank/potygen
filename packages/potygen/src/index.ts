export {
  AstTag,
  AnyCastTag,
  AnyTypeTag,
  ArrayColumnIndexTag,
  ArrayConstructorTag,
  ArrayIndexRangeTag,
  ArrayIndexTag,
  BeginTag,
  BinaryExpressionTag,
  BinaryOperatorTag,
  BitStringTag,
  BooleanTag,
  CaseSimpleTag,
  CaseTag,
  CastableDataTypeTag,
  CastTag,
  CollateTag,
  ColumnsTag,
  ColumnTag,
  CombinationTag,
  CommentTag,
  ComparationArrayInclusionTypeTag,
  ComparationArrayOperatorTag,
  ComparationArrayTypeTag,
  ComparationArrayInclusionTag,
  ComparationArrayTag,
  ExistsTag,
  CompositeAccessTag,
  ConflictConstraintTag,
  ConflictTag,
  ConflictTargetTag,
  ConstantTag,
  CountTag,
  CTENameTag,
  CTETag,
  CTEValuesListTag,
  CTEValuesTag,
  CustomQuotedStringTag,
  DataTypeTag,
  DefaultTag,
  DeleteTag,
  DimensionTag,
  DistinctTag,
  DollarQuotedStringTag,
  DoNothingTag,
  DoUpdateTag,
  ElseTag,
  EmptyLeafSqlTag,
  EmptyLeafTag,
  EscapeStringTag,
  ExpressionListTag,
  ExpressionTag,
  ExtractFieldTag,
  ExtractTag,
  FilterTag,
  FromListItemTag,
  FromListTag,
  FromTag,
  FunctionArgTag,
  FunctionTag,
  GroupByTag,
  HavingTag,
  HexademicalStringTag,
  IdentifierTag,
  InsertTag,
  JoinOnTag,
  JoinTag,
  JoinTypeTag,
  JoinUsingTag,
  LeafSqlTag,
  LimitAllTag,
  LimitTag,
  NamedSelectTag,
  NodeSqlTag,
  NodeTag,
  NullTag,
  NumberTag,
  OffsetTag,
  OperatorExpressionTag,
  OrderByItemTag,
  OrderByTag,
  OrderDirectionTag,
  ParameterTag,
  ParameterPickTag,
  PgCastTag,
  QualifiedIdentifierTag,
  QueryTag,
  QuotedIdentifierTag,
  ReturningListItemTag,
  ReturningTag,
  RollbackTag,
  RowTag,
  RowKeywardTag,
  SavepointTag,
  SelectListItemTag,
  SelectListTag,
  SelectParts,
  SelectTag,
  SetItemTag,
  SetArrayItemTag,
  SetListTag,
  SetMapTag,
  SetTag,
  SqlTag,
  StarIdentifierTag,
  StarTag,
  StringTag,
  TableTag,
  TableWithJoinTag,
  Tag,
  TernaryExpressionTag,
  TernaryOperatorTag,
  TernarySeparatorTag,
  ArrayTypeTag,
  TypeTag,
  UnaryExpressionTag,
  UnaryOperatorTag,
  UnquotedIdentifierTag,
  UpdateFromTag,
  UpdateTag,
  UsingTag,
  ValuesListTag,
  ValuesTag,
  WhenTag,
  WhereTag,
  WithTag,
  WrappedExpressionTag,
  SqlName,
} from './grammar.types';

export {
  isAnyCast,
  isAnyType,
  isArrayColumnIndex,
  isArrayConstructor,
  isArrayIndex,
  isArrayIndexRange,
  isAs,
  isBinaryExpression,
  isBinaryOperator,
  isBitString,
  isBoolean,
  isCase,
  isCaseSimple,
  isCast,
  isCastableDataType,
  isCollate,
  isColumn,
  isColumns,
  isCombination,
  isComment,
  isComparationArrayInclusionType,
  isComparationArrayOperator,
  isComparationArrayType,
  isComparationArrayInclusion,
  isComparationArray,
  isExists,
  isCompositeAccess,
  isConflict,
  isConflictConstraint,
  isConflictTarget,
  isConflictTargetIndex,
  isConstant,
  isCount,
  isCTE,
  isCTEName,
  isCTEValues,
  isCTEValuesList,
  isCustomQuotedString,
  isDataType,
  isDefault,
  isDelete,
  isDistinct,
  isDollarQuotedString,
  isDoNothing,
  isDoUpdate,
  isElse,
  isEmptyLeaf,
  isEscapeString,
  isExpression,
  isExpressionList,
  isFilter,
  isFrom,
  isFromList,
  isFromListItem,
  isFunction,
  isFunctionArg,
  isGroupBy,
  isHaving,
  isHexademicalString,
  isIdentifier,
  isInsert,
  isInteger,
  isJoin,
  isJoinOn,
  isJoinType,
  isJoinUsing,
  isLeaf,
  isLimit,
  isLimitAll,
  isNamedSelect,
  isNode,
  isNull,
  isNumber,
  isOffset,
  isOperatorExpression,
  isOrderBy,
  isOrderByItem,
  isOrderDirection,
  isParameter,
  isParameterPick,
  isPgCast,
  isQualifiedIdentifier,
  isQuotedIdentifier,
  isReturning,
  isReturningListItem,
  isRow,
  isRowKeyward,
  isSelect,
  isSelectList,
  isSelectListItem,
  isSet,
  isSetItem,
  isSetArrayItem,
  isSetList,
  isSetMap,
  isStar,
  isStarIdentifier,
  isString,
  isTable,
  isTableWithJoin,
  isTernaryExpression,
  isArrayType,
  isUnaryExpression,
  isUnaryOperator,
  isUnquotedIdentifier,
  isUpdate,
  isUpdateFrom,
  isUsing,
  isValues,
  isValuesList,
  isWhen,
  isWhere,
  isWith,
  isWrappedExpression,
} from './grammar.guards';

export {
  chunk,
  first,
  groupBy,
  identity,
  initial,
  isDiffBy,
  isEmpty,
  isEqual,
  isNil,
  isObject,
  isUnique,
  isUniqueBy,
  last,
  orderBy,
  tail,
  toMilliseconds,
  range,
  findLastIndex,
} from './util';

export { parser, partialParser } from './grammar';

export {
  TypeString,
  TypeNumber,
  TypeBoolean,
  TypeDate,
  TypeNull,
  TypeJson,
  TypeUnknown,
  TypeLoadCoalesce,
  TypeLoadRecord,
  TypeLoadFunction,
  TypeLoadColumn,
  TypeLoadStar,
  TypeLoadFunctionArgument,
  TypeLoadOperator,
  TypeLoadNamed,
  TypeLoadArray,
  TypeLoadAsArray,
  TypeAny,
  TypeLoadArrayItem,
  TypeContext,
  TypeLiteral,
  BaseTypeLoad,
  BaseType,
  TypeNullable,
  TypeLoadUnion,
  TypeBuffer,
  TypeUnion,
  TypeArray,
  Type,
  TypeOrLoad,
  Result,
  Param,
  ParamPick,
  Source,
  SourceTable,
  SourceQuery,
  QueryInterface,
  TypeLoadObjectLiteral,
  TypeObjectLiteral,
  TypeLoadCompositeAccess,
  TypeComposite,
  TypeLoadOptional,
  TypeOptional,
  TypeLoadColumnCast,
  OperatorVariantPart,
  OperatorVariant,
  TypeName,
} from './query-interface.types';

export {
  isType,
  isTypeString,
  isTypeNumber,
  isTypeBoolean,
  isTypeDate,
  isTypeNull,
  isTypeJson,
  isTypeUnknown,
  isTypeLoadRecord,
  isTypeLoadFunction,
  isTypeLoadColumn,
  isTypeLoadStar,
  isTypeLoadFunctionArgument,
  isTypeLoadOperator,
  isTypeLoadNamed,
  isTypeCoalesce,
  isTypeLoadArray,
  isTypeLoadUnion,
  isTypeArray,
  isTypeNullable,
  isTypeLiteral,
  isTypeUnion,
  isTypeLoadObjectLiteral,
  isTypeAny,
  isTypeLoadArrayItem,
  isTypeLoadAsArray,
  isTypeObjectLiteral,
  isTypeComposite,
  isTypeLoadCompositeAccess,
  isTypeLoadOptional,
  isTypeOptional,
  isTypeLoadColumnCast,
  isSourceTable,
  isSourceQuery,
  isSourceValues,
  isTypeEqual,
  isTypeLoad,
} from './query-interface.guards';

export { SqlInterface, QueryConfig, QuerySource, SqlResult, SqlDatabase, Query, MapQuery, Json } from './sql.types';
export { oneResult, maybeOneResult, mapResult } from './sql.helpers';
export {
  toParams,
  toQueryInterface,
  toConstantBinaryOperatorVariant,
  toAliasedPgType,
  toPgTypeConstant,
  toSources,
  toQueryFrom,
  toQueryResults,
} from './query-interface';
export { sql, toQuery, toQueryConfig } from './sql';
export {
  DatabaseError,
  isDatabaseError,
  PotygenError,
  PotygenDatabaseError,
  PotygenNotFoundError,
  LoadError,
  ParseError,
  ParsedTypescriptFileLoadError,
  ParsedSqlFileLoadError,
} from './errors';

export {
  DataTable,
  DataFunction,
  DataEnum,
  DataView,
  DataComposite,
  LoadedDataTable,
  LoadedDataComposite,
  DataViewRaw,
  DataViewParsed,
  LoadedDataView,
  LoadedDataFunction,
  LoadedDataEnum,
  Data,
  LoadedData,
  LoadedParam,
  LoadedResult,
  LoadedQueryInterface,
  LoadedFunction,
  LoadedComposite,
  LoadedSourceTable,
  LoadedSourceQuery,
  LoadedSourceView,
  LoadedSourceValues,
  LoadedSource,
  LoadedContext,
  TemplateTagQuery,
  ParsedTypescriptFile,
  ParsedSqlFile,
  ParsedFile,
  LoadedTemplateTagQuery,
  LoadedTypescriptFile,
  LoadedSqlFile,
  LoadedFile,
  Logger,
  LoadContext,
  QualifiedName,
  Path,
  PathItem,
  CompletionEntry,
  InfoContext,
  InfoLoadedQuery,
  QuickInfo,
  Cache,
} from './load.types';

export {
  isDataTable,
  isDataFunction,
  isDataEnum,
  isLoadedDataTable,
  isLoadedDataFunction,
  isLoadedDataEnum,
  isLoadedDataView,
  isLoadedDataComposite,
  isLoadedSourceView,
  isLoadedSourceTable,
  isLoadedSourceValues,
  isLoadedSource,
  isLoadedSourceUnknown,
} from './load.guards';

export {
  toLoadedContext,
  loadQueryInterfacesData,
  toLoadedQueryInterface,
  loadData,
  loadAllData,
  extractDataSources,
  filterUnknownLoadedContext,
  throwOnUnknownLoadedContext,
} from './load';

export { LRUCache } from './inspect/cache';
export { closestParent, closestParentPath, toPath } from './inspect/path';
export { completionAtOffset, toInfoContext, quickInfoAtOffset, inspectError } from './inspect/inspect';
