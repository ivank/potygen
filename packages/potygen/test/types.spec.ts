import { Client } from 'pg';
import { toLoadedQueryInterface, Query, SqlInterface, sql, toQueryInterface } from '../src';
import { testDb } from './helpers';
let db: Client;

describe('Query Interface', () => {
  beforeAll(async () => {
    db = testDb();
    await db.connect();
  });

  afterAll(async () => {
    await db.end();
  });

  it.each<[string, Query<SqlInterface<unknown[]>>]>([
    ['aclitem', sql`SELECT 'pg_monitor=r*/pg_read_all_stats'::aclitem`],
    ['cid', sql`SELECT '1'::cid`],
    ['array strings', sql`SELECT ARRAY['one', 'two']`],
    ['array enums', sql`SELECT ARRAY['Pending', 'Done']::account_levelisation_state[]`],
    ['macaddr', sql`SELECT '08:00:2b:01:02:03'::macaddr`],
    ['macaddr8', sql`SELECT '08:00:2b:01:02:03:04:05'::macaddr8`],
    ['pg_lsn', sql`SELECT '16/B374D848'::pg_lsn`],
    ['tid', sql`SELECT '(42,9)'::tid`],
    ['uuid', sql`SELECT 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A11'::uuid`],
    ['xid', sql`SELECT '1'::xid`],
    ['interval', sql`SELECT '1 day + 1 hour'::interval`],
    ['bytea', sql`SELECT E'\\\\000'::bytea`],
    ['name', sql`SELECT 'day'::name`],
    ['money', sql`SELECT '$1'::money`],
    ['regtype', sql`SELECT '1'::regtype`],
    ['regclass', sql`SELECT '1'::regclass`],
    ['regconfig', sql`SELECT '1'::regconfig`],
    ['regdictionary', sql`SELECT '1'::regdictionary`],
    ['regnamespace', sql`SELECT '1'::regnamespace`],
    ['regoper', sql`SELECT '1'::regoper`],
    ['regoperator', sql`SELECT '1'::regoperator`],
    ['regproc', sql`SELECT '1'::regproc`],
    ['regprocedure', sql`SELECT '1'::regprocedure`],
    ['regrole', sql`SELECT '1'::regrole`],
    ['regtype', sql`SELECT '1'::regtype`],
    ['box', sql`SELECT '(1,2),(3,4)'::box`],
    ['circle', sql`SELECT '<(1,2),3>'::circle`],
    ['line', sql`SELECT '{1,-1,1}'::line`],
    ['lseg', sql`SELECT '(1,2,3,4)'::lseg`],
    ['point', sql`SELECT '(1,2)'::point`],
    ['tsrange', sql`SELECT '[2010-01-01 14:30, 2010-01-01 15:30)'::tsrange`],
    ['numrange', sql`SELECT '[1,2)'::numrange`],
    ['daterange', sql`SELECT '[2010-01-01, 2010-01-02)'::daterange`],
    ['txid_snapshot', sql`SELECT '10:20:10,14,15'::txid_snapshot`],
    ['bit', sql`SELECT E'1'::bit`],
    ['varbit', sql`SELECT E'111'::varbit`],
    ['bpchar', sql`SELECT 'A'::bpchar`],
    ['cidr', sql`SELECT '1.1.1.1/32'::cidr`],
    ['inet', sql`SELECT '::10.2.3.4'::inet`],
    ['int4range', sql`SELECT '[1,2)'::int4range`],
    ['int8range', sql`SELECT '[1,2)'::int8range`],
    ['int2vector', sql`SELECT '1'::int2vector`],
    ['cstring', sql`SELECT '1'::cstring`],
    ['numeric', sql`SELECT '1.2'::numeric`],
    ['path', sql`SELECT '[(1,2),(2,3),(3,4)]'::path`],
    ['polygon', sql`SELECT '((1,2),(2,3),(1,2))'::polygon`],
  ])('Should load type for %s sql', async (_, sql) => {
    const queryInterface = toQueryInterface(sql().ast);
    const loadedQueryInterface = toLoadedQueryInterface([])(queryInterface);
    expect((await sql(db, {}))[0]).toMatchSnapshot();
    expect(loadedQueryInterface).toMatchSnapshot();
  });
});
