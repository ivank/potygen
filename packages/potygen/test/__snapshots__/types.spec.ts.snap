// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query Interface Should load type for aclitem sql 1`] = `
Object {
  "aclitem": "pg_monitor=r*/pg_read_all_stats",
}
`;

exports[`Query Interface Should load type for aclitem sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "aclitem",
      "type": Object {
        "postgresType": "aclitem",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for bit sql 1`] = `
Object {
  "bit": "1",
}
`;

exports[`Query Interface Should load type for bit sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "bit",
      "type": Object {
        "postgresType": "bit",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for box sql 1`] = `
Object {
  "box": "(3,4),(1,2)",
}
`;

exports[`Query Interface Should load type for box sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "box",
      "type": Object {
        "postgresType": "box",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for bpchar sql 1`] = `
Object {
  "bpchar": "A",
}
`;

exports[`Query Interface Should load type for bpchar sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "bpchar",
      "type": Object {
        "postgresType": "bpchar",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for bytea sql 1`] = `
Object {
  "bytea": Object {
    "data": Array [
      0,
    ],
    "type": "Buffer",
  },
}
`;

exports[`Query Interface Should load type for bytea sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "bytea",
      "type": Object {
        "postgresType": "bytea",
        "type": 0,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for cid sql 1`] = `
Object {
  "cid": "1",
}
`;

exports[`Query Interface Should load type for cid sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "cid",
      "type": Object {
        "postgresType": "cid",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for cidr sql 1`] = `
Object {
  "cidr": "1.1.1.1/32",
}
`;

exports[`Query Interface Should load type for cidr sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "cidr",
      "type": Object {
        "postgresType": "cidr",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for circle sql 1`] = `
Object {
  "circle": Object {
    "radius": 3,
    "x": 1,
    "y": 2,
  },
}
`;

exports[`Query Interface Should load type for circle sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "circle",
      "type": Object {
        "items": Array [
          Object {
            "name": "x",
            "type": Object {
              "postgresType": "float4",
              "type": 3,
            },
          },
          Object {
            "name": "y",
            "type": Object {
              "postgresType": "float4",
              "type": 3,
            },
          },
          Object {
            "name": "radius",
            "type": Object {
              "postgresType": "float4",
              "type": 3,
            },
          },
        ],
        "postgresType": "circle",
        "type": 13,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for cstring sql 1`] = `
Object {
  "cstring": "1",
}
`;

exports[`Query Interface Should load type for cstring sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "cstring",
      "type": Object {
        "postgresType": "cstring",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for daterange sql 1`] = `
Object {
  "daterange": "[2010-01-01,2010-01-02)",
}
`;

exports[`Query Interface Should load type for daterange sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "daterange",
      "type": Object {
        "postgresType": "daterange",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for inet sql 1`] = `
Object {
  "inet": "::10.2.3.4",
}
`;

exports[`Query Interface Should load type for inet sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "inet",
      "type": Object {
        "postgresType": "inet",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for int2vector sql 1`] = `
Object {
  "int2vector": "1",
}
`;

exports[`Query Interface Should load type for int2vector sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "int2vector",
      "type": Object {
        "postgresType": "int2vector",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for int4range sql 1`] = `
Object {
  "int4range": "[1,2)",
}
`;

exports[`Query Interface Should load type for int4range sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "int4range",
      "type": Object {
        "postgresType": "int4range",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for int8range sql 1`] = `
Object {
  "int8range": "[1,2)",
}
`;

exports[`Query Interface Should load type for int8range sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "int8range",
      "type": Object {
        "postgresType": "int8range",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for interval sql 1`] = `
Object {
  "interval": PostgresInterval {
    "days": 1,
    "hours": 1,
  },
}
`;

exports[`Query Interface Should load type for interval sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "interval",
      "type": Object {
        "items": Array [
          Object {
            "name": "years",
            "type": Object {
              "postgresType": "float4",
              "type": 3,
            },
          },
          Object {
            "name": "months",
            "type": Object {
              "postgresType": "float4",
              "type": 3,
            },
          },
          Object {
            "name": "days",
            "type": Object {
              "postgresType": "float4",
              "type": 3,
            },
          },
          Object {
            "name": "hours",
            "type": Object {
              "postgresType": "float4",
              "type": 3,
            },
          },
          Object {
            "name": "minutes",
            "type": Object {
              "postgresType": "float4",
              "type": 3,
            },
          },
          Object {
            "name": "seconds",
            "type": Object {
              "postgresType": "float4",
              "type": 3,
            },
          },
          Object {
            "name": "milliseconds",
            "type": Object {
              "postgresType": "float4",
              "type": 3,
            },
          },
        ],
        "postgresType": "interval",
        "type": 13,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for line sql 1`] = `
Object {
  "line": "{1,-1,1}",
}
`;

exports[`Query Interface Should load type for line sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "line",
      "type": Object {
        "postgresType": "line",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for lseg sql 1`] = `
Object {
  "lseg": "[(1,2),(3,4)]",
}
`;

exports[`Query Interface Should load type for lseg sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "lseg",
      "type": Object {
        "postgresType": "lseg",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for macaddr sql 1`] = `
Object {
  "macaddr": "08:00:2b:01:02:03",
}
`;

exports[`Query Interface Should load type for macaddr sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "macaddr",
      "type": Object {
        "postgresType": "macaddr",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for macaddr8 sql 1`] = `
Object {
  "macaddr8": "08:00:2b:01:02:03:04:05",
}
`;

exports[`Query Interface Should load type for macaddr8 sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "macaddr8",
      "type": Object {
        "postgresType": "macaddr8",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for money sql 1`] = `
Object {
  "money": "$1.00",
}
`;

exports[`Query Interface Should load type for money sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "money",
      "type": Object {
        "postgresType": "money",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for name sql 1`] = `
Object {
  "name": "day",
}
`;

exports[`Query Interface Should load type for name sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "name",
      "type": Object {
        "postgresType": "name",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for numeric sql 1`] = `
Object {
  "numeric": "1.2",
}
`;

exports[`Query Interface Should load type for numeric sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "decimal",
      "type": Object {
        "postgresType": "decimal",
        "type": 3,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for numrange sql 1`] = `
Object {
  "numrange": "[1,2)",
}
`;

exports[`Query Interface Should load type for numrange sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "numrange",
      "type": Object {
        "postgresType": "numrange",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for path sql 1`] = `
Object {
  "path": "[(1,2),(2,3),(3,4)]",
}
`;

exports[`Query Interface Should load type for path sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "path",
      "type": Object {
        "postgresType": "path",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for pg_lsn sql 1`] = `
Object {
  "pg_lsn": "16/B374D848",
}
`;

exports[`Query Interface Should load type for pg_lsn sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "pg_lsn",
      "type": Object {
        "postgresType": "pg_lsn",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for point sql 1`] = `
Object {
  "point": Object {
    "x": 1,
    "y": 2,
  },
}
`;

exports[`Query Interface Should load type for point sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "point",
      "type": Object {
        "items": Array [
          Object {
            "name": "x",
            "type": Object {
              "postgresType": "float4",
              "type": 3,
            },
          },
          Object {
            "name": "y",
            "type": Object {
              "postgresType": "float4",
              "type": 3,
            },
          },
        ],
        "postgresType": "point",
        "type": 13,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for polygon sql 1`] = `
Object {
  "polygon": "((1,2),(2,3),(1,2))",
}
`;

exports[`Query Interface Should load type for polygon sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "polygon",
      "type": Object {
        "postgresType": "polygon",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for regclass sql 1`] = `
Object {
  "regclass": "1",
}
`;

exports[`Query Interface Should load type for regclass sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "regclass",
      "type": Object {
        "postgresType": "regclass",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for regconfig sql 1`] = `
Object {
  "regconfig": "1",
}
`;

exports[`Query Interface Should load type for regconfig sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "regconfig",
      "type": Object {
        "postgresType": "regconfig",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for regdictionary sql 1`] = `
Object {
  "regdictionary": "1",
}
`;

exports[`Query Interface Should load type for regdictionary sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "regdictionary",
      "type": Object {
        "postgresType": "regdictionary",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for regnamespace sql 1`] = `
Object {
  "regnamespace": "1",
}
`;

exports[`Query Interface Should load type for regnamespace sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "regnamespace",
      "type": Object {
        "postgresType": "regnamespace",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for regoper sql 1`] = `
Object {
  "regoper": "1",
}
`;

exports[`Query Interface Should load type for regoper sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "regoper",
      "type": Object {
        "postgresType": "regoper",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for regoperator sql 1`] = `
Object {
  "regoperator": "1",
}
`;

exports[`Query Interface Should load type for regoperator sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "regoperator",
      "type": Object {
        "postgresType": "regoperator",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for regproc sql 1`] = `
Object {
  "regproc": "1",
}
`;

exports[`Query Interface Should load type for regproc sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "regproc",
      "type": Object {
        "postgresType": "regproc",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for regprocedure sql 1`] = `
Object {
  "regprocedure": "1",
}
`;

exports[`Query Interface Should load type for regprocedure sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "regprocedure",
      "type": Object {
        "postgresType": "regprocedure",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for regrole sql 1`] = `
Object {
  "regrole": "1",
}
`;

exports[`Query Interface Should load type for regrole sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "regrole",
      "type": Object {
        "postgresType": "regrole",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for regtype sql 1`] = `
Object {
  "regtype": "1",
}
`;

exports[`Query Interface Should load type for regtype sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "regtype",
      "type": Object {
        "postgresType": "regtype",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for regtype sql 3`] = `
Object {
  "regtype": "1",
}
`;

exports[`Query Interface Should load type for regtype sql 4`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "regtype",
      "type": Object {
        "postgresType": "regtype",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for reltime sql 1`] = `
Object {
  "reltime": "1 day",
}
`;

exports[`Query Interface Should load type for reltime sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "reltime",
      "type": Object {
        "postgresType": "reltime",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for smgr sql 1`] = `
Object {
  "smgr": "magnetic disk",
}
`;

exports[`Query Interface Should load type for smgr sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "smgr",
      "type": Object {
        "postgresType": "smgr",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for tid sql 1`] = `
Object {
  "tid": "(42,9)",
}
`;

exports[`Query Interface Should load type for tid sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "tid",
      "type": Object {
        "postgresType": "tid",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for tsrange sql 1`] = `
Object {
  "tsrange": "[\\"2010-01-01 14:30:00\\",\\"2010-01-01 15:30:00\\")",
}
`;

exports[`Query Interface Should load type for tsrange sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "tsrange",
      "type": Object {
        "postgresType": "tsrange",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for txid_snapshot sql 1`] = `
Object {
  "txid_snapshot": "10:20:10,14,15",
}
`;

exports[`Query Interface Should load type for txid_snapshot sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "txid_snapshot",
      "type": Object {
        "postgresType": "txid_snapshot",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for uuid sql 1`] = `
Object {
  "uuid": "a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11",
}
`;

exports[`Query Interface Should load type for uuid sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "uuid",
      "type": Object {
        "postgresType": "uuid",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for varbit sql 1`] = `
Object {
  "varbit": "111",
}
`;

exports[`Query Interface Should load type for varbit sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "varbit",
      "type": Object {
        "postgresType": "varbit",
        "type": 2,
      },
    },
  ],
}
`;

exports[`Query Interface Should load type for xid sql 1`] = `
Object {
  "xid": "1",
}
`;

exports[`Query Interface Should load type for xid sql 2`] = `
Object {
  "params": Array [],
  "results": Array [
    Object {
      "name": "xid",
      "type": Object {
        "postgresType": "xid",
        "type": 2,
      },
    },
  ],
}
`;
